core/lib/libtomcrypt/sm2-dsa.c

```diff
@@ -121,8 +121,8 @@ TEE_Result crypto_acipher_sm2_dsa_sign(uint32_t algo,
 	/* Step A7: convert (r, s) to binary for output */
 
 	memset(sig, 0, 64);
-	mp_to_unsigned_bin(r, sig);
-	mp_to_unsigned_bin(s, sig + 32);
+	mp_to_unsigned_bin(r, sig + 32 - mp_unsigned_bin_size(r));
```

Comment:
The layout of sig need a short recapture. Please add a line or two of comments describing the thing with `+ 32 -  mp_unsigned_bin_size(r)` etc.

Reply:
OK. The purpose of `32 - mp_unsigned_bin_size(x)` thing is to fill the whole destination size (32 bytes) with the bignum, whatever the actual size of the bignum. Since we convert to big endian, the zero padding has to occur on the left (lower addresses). In fact the `mp_to_unsigned_bin()` interface is quite bad IMO, because it does not have the size of the target buffer -- Mbed TLS has `mbedtls_mpi_write_binary(const mbedtls_mpi *X, unsigned char *buf, size_t buflen)` which is much better in that respect.

How about adding a helper macro in `core/lib/libtomcrypt/src/headers/tomcrypt_private.h`?
```C
#define mp_to_unsigned_bin2(a, b, c) do { \
                void *_a = (a); \
                mp_to_unsigned_bin(_a, (b) + (c) - mp_unsigned_bin_size(_a)); \
        } while(0)
```